app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -362,6 +362,29 @@ def enabled_dns_support?(vpc)
     vpc.describe_attribute(:attribute => 'enableDnsSupport', :vpc_id => vpc.vpc_id).enable_dns_support.value
   end
 
+  def enabled_internet_gateways?(vpc)
+    igw_ids = vpc.internet_gateways.map(&:internet_gateway_id)
+
+    unless igw_ids.empty? # Gateway attached
+      _log.debug("Found a gateway [#{igw_ids.first}] on VPC [#{vpc.id}]")
+
+      # Make sure gateway has vaild route
+      route_tables = vpc.route_tables.select do |route_table|
+        route_table.routes.any? { |route| route.gateway_id == igw_ids.first }
```

Comment:
Can't there be more than one gateway?  It looks like you're only checking if the route's gateway matches the first gateway id - what if the route's gateway matches another one of the gateways on the VPC?

Reply:
It's a one to one relationship between VPC and gateway.